server/src/main/java/com/kentaromatt/todoapi/ToDoController.java

```diff
@@ -32,6 +35,28 @@
         }};
     }
 
+    @PostMapping(path = "/todos")
+    public @ResponseBody HashMap<String, String> addToDo(@RequestBody HashMap<String, String> requestMap) {
+        if (!requestMap.containsKey("description") || requestMap.get("description") == "") {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Description is required");
+        }
+        String description = requestMap.get("description");
+        try {
+            LocalDate dueDate = LocalDate.parse(requestMap.get("dueDate"));
+            ToDo newToDo = new ToDo(description, dueDate);
+            repository.save(newToDo);
+            return new HashMap<String, String>(){{
+                put("id", newToDo.getId().toString());
+                put("description", newToDo.getDescription());
+                put("dueDate", newToDo.getDueDate().toString());
+                put("isComplete", newToDo.getIsComplete().toString());
+            }};
+        } catch (Exception e) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "dueDate must be in format YYYY-MM-DD");
```

Comment:
Do you think this error message is a bit too specific? I'm thinking it could be a database error on line 47, `repository.save(newToDo);`.

Reply:
Good call! I refactored this to handle the dueDate exception separately, and then create a new todo after description and dueDate have been verified